packages/discovery-react-components/src/components/SearchFacets/utils/buildAggregationQuery.ts

```diff
@@ -1,12 +1,15 @@
-import DiscoveryV2 from 'ibm-watson/discovery/v2';
+import { QueryAggregationWithName } from './searchFacetInterfaces';
 
-export const buildAggregationQuery = (
-  configuration: DiscoveryV2.QueryTermAggregation[]
-): string => {
-  const aggregation = configuration.map(term => {
-    return term.count
-      ? 'term(' + term.field + ',count:' + term.count + ')'
-      : 'term(' + term.field + ')';
+export const buildAggregationQuery = (configuration: QueryAggregationWithName[]): string => {
+  const aggregation = configuration!.map(term => {
+    const termCount = term.count ? ',count:' + term.count : '';
+    const termName = term.name ? ',name:' + term.name : '';
+    let nestedTypeTermAgg = '';
+    if (term.field!.includes('enriched_') && term.field!.includes('entities.text')) {
```

Comment:
can we type the `QueryAggregationWithName` to make `field` required? that way we don't have `!` in production code

Reply:
If I can figure out a better way around the issue with `DiscoveryV2.QueryAggregation` not having field required but that being in the response from the search client, then yes. I agree that would be better.

Reply:
maybe take a look at type guards? that should enforce a type as long as `QueryTermAggregation` extends `QueryAggregation` appropriately -> https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
Ah nice, thank you!